test/server/test_fxa_auth.py

```diff
@@ -0,0 +1,78 @@
+import urlparse
+import urllib
+from clientlib import ScreenshotsClient
+from urlparse import urljoin
+import random
+import requests
+import time
+from requests import HTTPError
+
+# Hack to make this predictable:
+random.seed(0)
+
+
+def get_url(url, expect=None):
+    result = urllib.urlopen(url)
+    if expect and result.getcode() != expect:
+        raise Exception("GET {} returned status {}".format(url, result.getcode()))
+    return result.read()
+
+# test we can change expiration and delete shots made previously
+def test_owned_shot_with_fxa_auth():
+    first_user = ScreenshotsClient()
+    second_user = ScreenshotsClient()
+
+    first_user.login();
+    shot_url = first_user.create_shot(docTitle="A_TEST_SITE_1")
+    shot_id = urlparse.urlsplit(shot_url).path.strip("/")
+    shot_page = first_user.read_shot(shot_url)
+    print(first_user.read_shot(shot_url)["clip_url"], shot_page["clip_url"])
+    assert first_user.read_shot(shot_url)["clip_content"] == shot_page["clip_content"]
+    assert "A_TEST_SITE_1" in shot_page["page"]
+
+
+    # Use first_user accountId to login to a second session
+    second_user.fxa_login(first_user.accountId)
+
+    # Read shot page to check if second_user can set expiration and delete shot
+    page = second_user.read_shot(shot_url)
+    second_user.set_expiration(shot_url, 1)
+    time.sleep(2)
+    expired_page = second_user.read_shot(shot_url)
+    # An expired page happens to be unparseable:
+    assert expired_page["clip_content"] is None
+    get_url(page["clip_url"], expect=404)
+    get_url(shot_url, expect=200)
+    second_user.delete_shot(shot_url)  # cleanup
+
+
+# test we cannot change expiration or delete shots made with a different deviceId/accountId
+def test_not_owned_shot():
+    first_user = ScreenshotsClient()
+    second_user = ScreenshotsClient()
+
+    first_user.login();
+    shot_url = first_user.create_shot(docTitle="A_TEST_SITE_2")
+    shot_id = urlparse.urlsplit(shot_url).path.strip("/")
+    shot_page = first_user.read_shot(shot_url)
+    print(first_user.read_shot(shot_url)["clip_url"], shot_page["clip_url"])
+    assert first_user.read_shot(shot_url)["clip_content"] == shot_page["clip_content"]
+    assert "A_TEST_SITE_2" in shot_page["page"]
+
+    second_user.login();
+    # Read !owned shot created in first_user session
+    page = second_user.read_shot(shot_url)
+
+    # set expiration on the shot should throw 404
+    try:
+        try:
+            second_user.set_expiration(shot_url, 1)
+        except HTTPError, e:
+            if e.response.status_code != 404:
+                raise
```

Comment:
This won't raise an error if it's a 200 response, only if it's >=300 and not 404.

You could just add `assert False, "set_expiration should fail"` after the second_user.set_expiration() call, or you could use the pytest pattern here: https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
Done